src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcileDatasetPidCommand.java

```diff
@@ -0,0 +1,181 @@
+        }
+        // We keep the old persistent identifier as AlternativePersistentIdentifier with storageLocationDesignator true.
+        // This keep the link the object store intact, without altering the files.
+        // IMHO: This command should also update the storage. First, maintenance of the storage becomes a mess if there is a counterintuitive layout.
+        // Second, it could occur that another object is minted with the old identifier and consequently we have a conflict in our storage system.
+        // We accept this risk for now since a super user can update the storage manually from old file path to the new one, and remove the AlternativeIdentifer form the database.
+        // This removed the old identifier totally from the system and avoids all side effects...
+        // @TODO: Move files on storage once the storage API support a move operation!
+        ctxt.em().merge(getDataset());
+        ctxt.em().flush();
+
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - Replaced old globalId `" + oldId + " with new globalId `" + getDataset().getGlobalId() + '`');
+        // notify all users with direct role assignments about the changed persistent identifier
+        List<RoleAssignment> ras = ctxt.roles().directRoleAssignments(getDataset());
+        for (RoleAssignment ra : ras) {
+            for (AuthenticatedUser au : ctxt.roleAssignees().getExplicitUsers(ctxt.roleAssignees().getRoleAssignee(ra.getAssigneeIdentifier()))) {
+                ctxt.notifications().sendNotification(au, new Timestamp(new Date().getTime()), UserNotification.Type.PIDRECONCILED,getDataset().getLatestVersion().getId(),"Persistent identifier changed!");
+            }
+        }
+
+        return getDataset();
+    }
+
+    private void reconcileFilePids(CommandContext ctxt) {
+        // remove datafile PIDs
+            try {
+                for (DataFile df : getDataset().getFiles()) {
+                    GlobalId oldPid=df.getGlobalId();
+                    PidProvider currentPidProvider = PidUtil.getPidProvider(oldPid.getProviderId());
+                    if (currentPidProvider.alreadyRegistered(df)) {
+                        currentPidProvider.deleteIdentifier(df);// delete it external
+                    }
+
+                    df.setGlobalId(null); // and remove it internally from data structure
+                    df.setGlobalIdCreateTime(null);
+                    df.setIdentifierRegistered(false);
+
+                    AlternativePersistentIdentifier api;
+                    api = new AlternativePersistentIdentifier();
+                    api.setProtocol(oldPid.getProtocol());
+                    api.setAuthority(oldPid.getAuthority());
+                    api.setIdentifier(oldPid.getIdentifier());
+                    api.setDvObject(df);
+                    api.setStorageLocationDesignator(true); // cf. Dataset#getIdentifierForFileStorage()
```

Comment:
same rare issue - if an altPID already exists that is the storage location designator